packages/playback/src/lib/eme/eme-manager.ts

```diff
@@ -89,5 +164,540 @@ export class EmeManager implements IEmeManager {
+
+      // Do we need this?? Can we just store the initData in the array of activeSessions?
+      // this.logger_.debug(
+      //   `Updating init data: previous(${type}, length: ${data.byteLength}) --> new(${type}, length: ${data.byteLength})`
+      // );
+    }
+
+    const sessionId = mediaKeySession.sessionId;
+
+    mediaKeySession.addEventListener('keystatuseschange', (event) =>
+      this.onKeyStatusesChange_(event as ExtendableEvent)
+    );
+    mediaKeySession.addEventListener('message', (event) => this.onSessionMessage_(event));
+
+    // Register callback for session closed Promise
+    mediaKeySession.closed.then(() => {
+      this.removeSession_(sessionId);
+      this.logger_.debug('EME Key Session closed. sessionId: ' + sessionId);
+      this.eventEmitter_.emitEvent(new KeySessionClosedEvent(sessionId));
+    });
+
+    const metadata = {
+      initData,
+      initDataType,
+      loaded: false,
+      type: sessionType,
+      session: mediaKeySession,
+    };
+
+    this.activeSessions_.set(mediaKeySession.sessionId, metadata);
+
+    mediaKeySession
+      .generateRequest(initDataType, initData)
+      .then(() => {
+        this.logger_.debug('EME: Session created.  SessionID: ' + sessionId);
+        this.eventEmitter_.emitEvent(new KeySessionCreatedEvent(sessionId));
+      })
+      .catch((error) => {
+        this.removeSession_(sessionId);
+        this.eventEmitter_.emitEvent(new ErrorEvent(new KeySessionCreateError(false, error as Error)));
+      });
+  }
+
+  /**
+   * Sets the server certificate on the active media keys.
+   * @param certificate The server certificate
+   * @returns an empty promise.
+   */
+  private async setServerCertificate_(certificate: Uint8Array): Promise<void> {
+    if (!certificate) {
+      this.logger_.warn('EME: There was attempt to set an invalid server certificate on the active media keys.');
+      return;
+    }
+
+    if (!this.activeMediaKeys_) {
+      this.logger_.warn('EME: There was attempt to set a server certificate when the media keys do not exist.');
+      return;
+    }
+
+    try {
+      const isSupported = await this.activeMediaKeys_?.setServerCertificate(certificate);
+
+      if (!isSupported) {
+        this.logger_.warn('EME: The key system does not support server certificates. Ignoring this certificate.');
+      }
+
+      return;
+    } catch (error) {
+      this.eventEmitter_.emitEvent(new ErrorEvent(new InvalidServerCertificateError(false, error as Error)));
+    }
+  }
+
+  /**
+   * Event to handle key status changes event on session.
+   * @param event The event containing the media key session
+   */
+  private onKeyStatusesChange_(event: ExtendableEvent): void {
+    const session = event.target as MediaKeySession;
+    const activeSession = this.activeSessions_.get(session.sessionId);
+    const keyStatusMap = session.keyStatuses;
+
+    let hasExpiredKeys = false;
+
+    keyStatusMap.forEach((status, keyId) => {
```

Comment:
There is some additional complexity here that will likely have to be accounted for with HDCP, specifically mapping keyIds to 'usable' status'.